repository/repository.go

```diff
@@ -0,0 +1,147 @@
+package repository
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/BorisMustakimov/TD1314/nextdate"
+	"github.com/BorisMustakimov/TD1314/task"
+	"github.com/jmoiron/sqlx"
+)
+
+const Limit = 50
+
+// определяем интерфейс для работы с задачами 
+type TaskRepository interface {
+	Create(task *task.Task) (int64, error)
+	SearchTasks(filter Filter, id string) ([]task.Task, error)
+	UpdateTask(task *task.Task) error
+	Delete(id string) error
+}
+
+type TaskRepo struct {
+	db *sqlx.DB
+}
+
+// используем для фильтрации задач
+type Filter struct {
+	ID     []string
+	Search string
+	Date   string
+}
+
+func NewTaskRepo(db *sqlx.DB) TaskRepository {
+	return &TaskRepo{db: db}
+}
+
+// создание задачи
+func (r *TaskRepo) Create(task *task.Task) (int64, error) {
+	res, err := r.db.Exec(
+		`INSERT INTO scheduler (date, title, comment, repeat) VALUES (?,?,?,?)`,
+		task.Date, task.Title, task.Comment, task.Repeat,
+	)
+	if err != nil {
+		return 0, err
```

Comment:
Лучше оборачивать ошибки, чтобы быстрее понимать, где она произошла
```suggestion
		return 0, fmr.Errorf("error while executing request: %w", err)
```